kori/app/services/product.py

```diff
@@ -15,6 +16,15 @@ def get_product(product_id: UUID, timestamp: Optional[datetime] = None) -> Produ
     return product_dao.get_product(product_id=product_id, timestamp=timestamp)
 
 
+def get_products_by_stores(store_ids: list[UUID], product_name: str | None = None) -> list[ProductWithStock]:
+    aggregated_products = store_product_dao.get_products_by_stores(store_ids)
+    stock_mapping = {product.product_id: product.total_stock for product in aggregated_products}
+    products = product_dao.get_products_by_name(list(stock_mapping.keys()), product_name)
```

Comment:
This would work, but could be better.

Would do this in the reverse manner starting from `ProductVersion` as its cheap than trying to aggregate the whole table.